src/http-adapter/index.js

```diff
@@ -0,0 +1,218 @@
+/***
+ * @module http-adapter
+ */
+
+export const HTTP_METHOD = Object.freeze({
+  GET: 'GET',
+  PATCH: 'PATCH',
+  POST: 'POST',
+  DELETE: 'DELETE'
+})
+
+/**
+ * Adapter which works with data over HTTP, based on the options that
+ * are passed to its constructor. This class is responsible for
+ * any CRUD operations that need to be carried out with your {@link https://jsonapi.org/ JSON:API} service
+ * from within the Backbone.Store.
+ */
+export default class HttpAdapter {
+  /**
+   * Create a new instance of the HttpAdapter
+   * @param {Object} [options] - HttpAdapter options
+   * @param {string} options.urlPrefix - defines the REST service that will be returning a {@link https://jsonapi.org/ JSON:API} response.
+   * @param {boolean} [options.serializeRequests=false] - Whether requests need to wait on the previous request to resolve, IE running requests serially.
+   * @param {Function} [options.requestInterceptor=(()=>{})] - A method to be executed before each request to manipulate or augment the request. A popular use case
+   * of this is to add headers to a request.
+   * @param {Function} [options.responseInterceptor=(()=>{})]  - A method to be executed on response to manipulate or handle the given response.
+   * A use case of this would be to add generic behavior or error reporting for certain status codes.
+   */
+  constructor (options = {}) {
+    this.urlPrefix = options.urlPrefix
+    this.serializeRequests = false
+    this._outstandingRequests = new Set()
+    this.requestInterceptor = options.requestInterceptor || (() => {})
+    this.responseInterceptor = options.responseInterceptor || (() => {})
+  }
+
+  /**
+   * Parses a JSON:API resource to a restful url.
```

Comment:
Another way to phrase this is as building a url from the type and id resource identifiers. https://jsonapi.org/format/#document-resource-identifier-objects